### PR TITLE
Make all diskio functions and structs crate-private

### PR DESCRIPTION
**Pull request by 0xPoe**
part of https://github.com/rust-lang/rustup/issues/2730

### CODE REVIEW

**Comment by 0xPoe** [comment]
cc: @rbtcollins @kinnison  Could you please take a look?

**Review by kinnison — APPROVED** [review]

**Comment by kinnison** [comment]
While this looks good, we'll hold off merging until we've sorted the slab allocator stuff.

**Comment by kinnison** [comment]
This now needs rebasing

**Comment by 0xPoe** [comment]
> This now needs rebasing

Resolved.

**Review by kinnison — APPROVED** [review]
I think this is okay.  @rbtcollins if you have no objections, I think this can merge.

File: src/diskio/mod.rs [hunk 1]
```diff
@@ -176,41 +176,41 @@ pub enum IncrementalFile {
...
     IncrementalFile(IncrementalFile),
 }
 
 /// The details of the IO operation
 #[derive(Debug)]
-pub struct Item {
+pub(crate) struct Item {
     /// The path to operate on
-    pub full_path: PathBuf,
+    pub(crate) full_path: PathBuf,
```
Comment (rbtcollins) [inline src/diskio/mod.rs:189]: Can we answer the question about whether we need this annotation first?

Comment (rbtcollins) [reply src/diskio/mod.rs:189]: answered on discord: without this unused attributes are not detected.

**Pull request merged**